TypeSystem/Schema/schema.h

```diff
@@ -1115,6 +1117,202 @@ struct LanguageSyntaxImpl<Language::JSON> final {
   // LCOV_EXCL_STOP
 };
 
+template <>
+struct LanguageSyntaxImpl<Language::TypeScript> final {
+  static std::string Header(const std::string&) {
+    // TODO(sompylasar): Implement common C++ and Current types in TypeScript.
+    return (
+      "// Generated by C5T/Current.\n"
+      // TODO(sompylasar): Extract common types into a separate reusable schema.
+      "declare namespace C5TCurrent {\n"
+      "  type Bool = boolean;\n"
+      "  type UInt8 = number;\n"
+      "  type UInt16 = number;\n"
+      "  type UInt32 = number;\n"
+      "  type UInt64 = number;\n"
+      "  type Int8 = number;\n"
+      "  type Int16 = number;\n"
+      "  type Int32 = number;\n"
+      "  type Int64 = number;\n"
+      "  type Char = string;\n"
+      "  type String = string;\n"
+      "  type Float = number;\n"
+      "  type Double = number;\n"
+      "  type Microseconds = number;\n"
+      "  type Milliseconds = number;\n"
+      "  type Optional<T> = T | null;\n"
+      "  type Pair<T_FIRST, T_SECOND> = [ T_FIRST, T_SECOND ];\n"
+      "  type Set<T> = T[];\n"
+      "  type UnorderedSet<T> = T[];\n"
+      "  type NonPrimitiveMap<T_KEY, T_VALUE> = Pair<T_KEY, T_VALUE>[];\n"
+      "  type NonPrimitiveUnorderedMap<T_KEY, T_VALUE> = Pair<T_KEY, T_VALUE>[];\n"
+      "  interface PrimitiveMap<T_VALUE> { [key: string]: T_VALUE; [key: number]: T_VALUE; }\n"
+      "  interface PrimitiveUnorderedMap<T_VALUE> { [key: string]: T_VALUE; [key: number]: T_VALUE; }\n"
+      "}\n"
+    );
+  }
+
+  static std::string Footer(const std::string&) { return ""; }
+
+  static std::string SanitizeTypeScriptSymbol(const std::string& unsanitized_name) {
+    // https://github.com/Microsoft/TypeScript/blob/2a6aacd0ef614a38b08ef712adc377c13648f373/doc/spec.md#2.2.1
+    static std::set<std::string> typescript_reserved_words{
+      // TypeScript keywords that are reserved and cannot be used as an Identifier:
+      "break", "case", " catch", "class", "const", "continue", "debugger", "default", "delete", "do",
+      "else", "enum", "export", "extends", "false", "finally", "for", "function", "if", "import", "in",
+      "instanceof", "new", "null", "return", "super", "switch", "this", "throw", "true", "try", "typeof",
+      "var", "void", "while", "with",
+      // TypeScript keywords that cannot be used as identifiers in strict mode code, but are otherwise not restricted:
+      "implements", "interface", "let", "package", "private", "protected", "public", "static", "yield",
+      // TypeScript keywords cannot be used as user defined type names, but are otherwise not restricted:
+      "any", "boolean", "number", "string", "symbol"
+    };
+    return typescript_reserved_words.count(unsanitized_name) ? "'" + unsanitized_name + "'" : unsanitized_name;
```

Comment:
Would be awesome to have tests for this too (as well as the end-to-end one). I didn't note quotes in the golden output above.